This won't work on OCaml 4.x, but why don't we just make this library OCaml 5.x only now? I'm only practically using this in Eio which is obviously 5.x only.

```
File "lib/uring/dune", line 8, characters 9-20:
8 |   (names uring_stubs)
^^^^^^^^^^^
(cd _build/default/lib/uring && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -Wall -fdiagnostics-color=always -g -I /home/opam/.opam/4.14/lib/ocaml -I /home/opam/.opam/4.14/lib/cstruct -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/optint -I include -o uring_stubs.o -c uring_stubs.c)
uring_stubs.c: In function 'ocaml_uring_submit':
uring_stubs.c:937:5: error: implicit declaration of function 'caml_unix_error'; did you mean 'unix_error'? [-Wimplicit-function-declaration]
937 |     caml_unix_error(-num, "io_uring_submit", Nothing);
|     ^~~~~~~~~~~~~~~
|     unix_error

```

--- a/dune-project
+++ b/dune-project
@@ -14,7 +14,7 @@
  (description "Bindings to the Linux io_uring kernel IO interfaces. See https://github.com/ocaml-multicore/eio for a higher-level API using this.")
  (depends
   (cstruct (>= 6.0.1))
-  (ocaml (>= 4.12.0))
+  (ocaml (>= 5.2.0))
   dune-configurator
   (lwt (and :with-test (>= 5.0.0)))
   (bechamel (and (>= 0.1.0) :with-test))

--- a/lib/uring/uring_stubs.c
+++ b/lib/uring/uring_stubs.c
@@ -941,6 +941,9 @@ value ocaml_uring_submit(value v_uring)
   caml_enter_blocking_section();
   int num = io_uring_submit(ring);
   caml_leave_blocking_section();
+  if (num < 0) {
+    caml_unix_error(-num, "io_uring_submit", Nothing);
+  }
   CAMLreturn (Val_int(num));
 }
 

--- a/uring.opam
+++ b/uring.opam
@@ -11,7 +11,7 @@ bug-reports: "https://github.com/ocaml-multicore/ocaml-uring/issues"
 depends: [
   "dune" {>= "3.0"}
   "cstruct" {>= "6.0.1"}
-  "ocaml" {>= "4.12.0"}
+  "ocaml" {>= "5.2.0"}
   "dune-configurator"
   "lwt" {with-test & >= "5.0.0"}
   "bechamel" {>= "0.1.0" & with-test}

